framework/PostProcessors/ImportanceRank.py

```diff
@@ -454,69 +275,64 @@ def inputToInternal(self, currentInp):
+          inputDict['targets'][targetP] = copy.copy(dataSet[targetP].values)
         else:
           self.raiseAnError(IOError,'Parameter ' + str(targetP) + ' is listed ImportanceRank postprocessor targets, but not found in the provided output!')
-      inputDict['metadata'] = currentInput.getAllMetadata()
+      inputDict['metadata'] = currentInput.getMeta(pointwise=True)
+      inputList = [inputDict]
     # get input from HistorySet DataObject
     if inType in ['HistorySet']:
+      dataSet = currentInput.asDataset()
       if self.pivotParameter is None:
         self.raiseAnError(IOError, self, 'Time-dependent importance ranking is requested (HistorySet) but no pivotParameter got inputted!')
-      inputs  = currentInput.getParametersValues('inputs',nodeId = 'ending')
-      outputs = currentInput.getParametersValues('outputs',nodeId = 'ending')
-      numSteps = len(outputs.values()[0].values()[0])
       self.dynamic = True
-      if self.pivotParameter not in currentInput.getParaKeys('output'):
-        self.raiseAnError(IOError, self, 'Pivot parameter ' + self.pivotParameter + ' has not been found in output space of data object '+currentInput.name)
-      pivotParameter = []
-      for step in range(len(outputs.values()[0][self.pivotParameter])):
-        currentSnapShot = [outputs[i][self.pivotParameter][step] for i in outputs.keys()]
-        if len(set(currentSnapShot)) > 1:
-          self.raiseAnError(IOError, self, 'Histories are not syncronized! Please, pre-process the data using Interfaced PostProcessor HistorySetSync!')
-        pivotParameter.append(currentSnapShot[-1])
-      inputDict = {'timeDepData':OrderedDict.fromkeys(pivotParameter,None)}
-      for step in range(numSteps):
-        inputDict['timeDepData'][pivotParameter[step]] = {'targets':{},'features':{}}
-        for targetP in self.targets:
-          if targetP in currentInput.getParaKeys('output') :
-            inputDict['timeDepData'][pivotParameter[step]]['targets'][targetP] = np.asarray([outputs[i][targetP][step] for i in outputs.keys()])
-          else:
-            self.raiseAnError(IOError, self, 'Target ' + targetP + ' has not been found in data object '+currentInput.name)
-        for feat in self.features:
-          if feat in currentInput.getParaKeys('input'):
-            inputDict['timeDepData'][pivotParameter[step]]['features'][feat] = np.asarray([inputs[i][feat][-1] for i in inputs.keys()])
-          else:
-            self.raiseAnError(IOError, self, 'Feature ' + feat + ' has not been found in data object '+currentInput.name)
-        inputDict['timeDepData'][pivotParameter[step]]['metadata'] = currentInput.getAllMetadata()
-
+      self.pivotValue = dataSet[self.pivotParameter].values
+      if not currentInput.checkIndexAlignment(indexesToCheck=self.pivotParameter):
+        self.raiseAnError(IOError, "The data provided by the data objects ", currentInput.name, " is not synchronized!")
```

Comment:
This message might be expanded to clarify what "synchronized" means; i.e., that the realizations' pivot parameters have unsynchronized pivot values, and that the HistorySetSync internal PP can fix it.

Reply:
updated.